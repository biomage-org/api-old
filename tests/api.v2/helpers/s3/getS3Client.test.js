const getS3Client = require('../../../../src/api.v2/helpers/s3/getS3Client');
const AWS = require('../../../../src/utils/requireAWS');

jest.mock('../../../../src/utils/requireAWS', () => ({
  S3: jest.fn((params) => ({ ...params })),
}));

describe('getS3Client', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Returns an S3 client with defau lt config values if not given any params', () => {
    const s3Client = getS3Client();

    expect(AWS.S3).toHaveBeenCalledTimes(1);

    const configParams = AWS.S3.mock.calls[0][0];

    expect(configParams).toMatchSnapshot();
    expect(s3Client).not.toBeUndefined();
  });

  it('Takes in params and return S3 client with those params', () => {
    const additionalParams = {
      region: 'us-east-1',
      endpointUrl: 'https://s3.biomage-cloud.com',
    };

    const s3Client = getS3Client(additionalParams);

    expect(AWS.S3).toHaveBeenCalledTimes(1);

    const configParams = AWS.S3.mock.calls[0][0];

    expect(configParams).toMatchSnapshot();
    expect(s3Client).not.toBeUndefined();
  });
});
