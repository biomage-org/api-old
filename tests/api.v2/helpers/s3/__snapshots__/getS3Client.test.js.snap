// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getS3Client Returns an S3 client with defau lt config values if not given any params 1`] = `
Object {
  "apiVersion": "2006-03-01",
  "region": "eu-west-1",
  "signatureVersion": "v4",
}
`;

exports[`getS3Client Takes in params and return S3 client with those params 1`] = `
Object {
  "apiVersion": "2006-03-01",
  "endpointUrl": "https://s3.biomage-cloud.com",
  "region": "us-east-1",
  "signatureVersion": "v4",
}
`;
